ThesesFR.js

```diff
@@ -79,24 +85,80 @@ function doWeb(doc, url) {
 }
 
 function scrape(doc, url) {
-	var translator = Zotero.loadTranslator('web');
-	translator.setTranslator('951c027d-74ac-47d4-a107-9c3069ab7b48'); // https://github.com/zotero/translators/blob/master/Embedded%20Metadata.js
-	translator.setDocument(doc);
-	translator.setHandler('itemDone', function (obj, item) {
-		// add Tags
-		var tags = ZU.xpath(doc, '//span[contains(@property, "dc:subject")]');
-		if (tags.length > 0) {
-			item.tags = [];
-			for (let tag of tags) {
-				item.tags.push(tag.textContent.trim());
-			}
+	const xmlDocumentUrl = `${url}.rdf`;
+	
+	// Each thesis record has an underlying .rdf file
+	Zotero.Utilities.HTTP.doGet(xmlDocumentUrl, function (text) {
+		const parser = new DOMParser();
+		const xmlDoc = parser.parseFromString(text, 'application/xml');
+
+		// Skiping invalid or empty RDF files : prevents crashes while importing multiple records
+		if (xmlDoc.getElementsByTagName('parsererror')[0] || xmlDoc.children[0].childElementCount === 0) {
+			Z.debug('Skiping: Invalid or empty RDF file.');
+			return;
 		}
-			
-		item.complete();
-	});
-	translator.getTranslatorObject(function (trans) {
-		trans.itemType = "thesis";
-		trans.doWeb(doc, url);
+		
+		// Importing XML namespaces for parsing purposes
+		const ns = {
+			bibo: 'http://purorg/ontology/bibo/',
+			dc: 'http://purl.org/dc/elements/1.1/',
+			dcterms: 'http://purl.org/dc/terms/',
+			foaf: 'http://xmlns.com/foaf/0.1/',
+			marcrel: 'http://www.loc.gov/loc.terms/relators/',
+			rdf: 'http://www.w3.org/1999/02/22-rdf-syntax-ns#'
+		};
+	
+		const title = ZU.xpathText(xmlDoc, '//dc:title', ns);
+		
+		// The case should't exists : preve;nts crashes while importing multiple records
+		if (!title) return;
```

Comment:
Let's take this out please or replace it with a `throw` -- if an item unexpectedly doesn't import, we would _want_ an error to be thrown rather than to continue silently, letting the user assume all selected items have been imported.